content/04-postgresql/13-short-guides/03-connection-uris.mdx

```diff
@@ -0,0 +1,201 @@
+
+## Providing the database name
+
+After the `hostspec`, the next piece of data is the database name.  While not true for all database management systems, with PostgreSQL, you must connect to a specific database when establishing a connection.
+
+The database name begins with a forward slash (`/`) and proceeds until either the end of the line or a question mark (`?`).  You must include the database name if you aren't relying on the default values.
+
+To connect to a database called `sales` hosted on a PostgreSQL server listening on `198.51.100.22:3333`, you could type:
+
+```
+postgresql://username:password@198.51.100.22:3333/sales
+```
+
+## Specifying additional parameters
+
+The last part of the connection URI is used to provide additional parameters for the connection.  The list of parameters is introduced by a leading question mark (`?`) and continues until the end of the line.
+
+Each parameter listed is defined as a key and value pair joined with an equals sign (`=`).  After the first parameter pair, each additional key-value pair is separated by an ampersand (`&`).
+
+For example, to specify that the client should apply a 10 second timeout for the connection we were previously defining, you could use:
+
+```
+postgresql://username:password@198.51.100.22:3333/sales?connect_timeout=10
+```
+
+If you wanted to provide additional parameters, you'd add them afterwards with an ampersand (`&`) between each pair.  For instance, we could additionally specify that we require SSL and want to connect only if the server is a primary in a replica set, we could additionally add:
+
+```
+postgresql://username:password@198.51.100.22:3333/sales?connect_timeout=10&sslmode=require&target_session_attrs=primary
+```
+
+The PostgreSQL documentation has a [full list of parameters](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS) that you can read to learn more.
+
+## Conclusion
+
+In this guide, we discussed what a PostgreSQL connection URI is, how to interpret the various components, how to construct your own URIs given a set of connection information.  Connection URIs encode all of the information required to connect to a given database within a single string.  Because of this flexibility and due to their wide adoption, understanding how to parse and construct those strings can be pretty helpful.
+
+<PrismaOutlinks>
+
+If you are using [Prisma to manage your PostgreSQL database](https://www.prisma.io/docs/concepts/database-connectors/postgresql), you need to set a connection URI within a `datasource` block in your [Prisma schema file](https://www.prisma.io/docs/concepts/components/prisma-schema).  You must provide a [connection URI for the `url` field](https://www.prisma.io/docs/concepts/database-connectors/postgresql#example) so that Prisma can connect to your database.
```

Comment:
The `datasource` and `url` look really weird in the output of the outlink. If they are required to be written that way then so be it. Just something I noticed.